docs: readability, spelling and other documentation stuff

OdinGG opened the pull request.

OdinGG committed:
docs: fix javadoc formatting

OdinGG committed:
docs: readability and other fixes in README

OdinGG committed:
docs: better readability in README

OdinGG committed:
docs: fix spelling error in README

OdinGG committed:
docs: fixed javadoc formatting

OdinGG committed:
docs: added javadocs html

OdinGG committed:
docs: remove javadoc html

OdinGG committed:
docs: added javadoc for package.

Atheer2104 commented on README.md:
```diff
@@ -9,9 +9,9 @@ This is a simple CI server for the [Assignment #2: Continuous Integration](https
 ## Running the project
 - Clone the repo and make sure you are inside the CI subfolder
-- Run **mvn test** to confirm that everything is working
+- Run `mvn test` to confirm that everything is working, the "githubPAT" file is required.
```
I think the user has to also know where to put the githubPAT file I mention that is supposed to be in the root directory of the repo

Atheer2104 reviewed (commented):
Just one small comment about the githubPat, otherwise looks good, you can squash and merge after that

Atheer2104 committed:
docs: improve documentation

docs: add missing descriptions

docs: add description of parameter & fix typos

docs: add description of the parameters

docs: fix typo

docs: fix typo

docs: fix wrong parameter name for runTests function

docs: fix wrong usage of @author in javadoc

docs: remove @throws when function does not throw anything

refactor: remove unused import

Atheer2104 reviewed (approved):
LGTM
